workbench-core/datasets-ui/src/components/CreateInternalDatasetForm.tsx

```diff
@@ -0,0 +1,282 @@
+            // TODO: add href
+            info={<Link>Info</Link>}
+            description="The Project administrators on the selected project will be able to manage access to the dataset. This cannot be changed later."
+            errorText={formErrors?.projectIdError}
+          >
+            <Select
+              data-testid="datasetProject"
+              selectedOption={
+                areProjectsLoading
+                  ? null
+                  : projects
+                      .map((p) => ({ label: p.name, value: p.id }))
+                      .filter((p) => p.value === formData?.projectId)[0] || null
+              }
+              loadingText="Loading Projects"
+              options={projects.map((p) => ({ label: p.name, value: p.id }))}
+              selectedAriaLabel={formData?.projectId}
+              onChange={({ detail: { selectedOption } }) => {
+                setFormData({ ...formData, projectId: selectedOption.value });
+                validateField<string | undefined, CreateDatasetForm, CreateDatasetFormValidation>(
+                  'projectId',
+                  setFormErrors,
+                  datasetProjectIdValidationRules,
+                  selectedOption.value
+                );
+              }}
+              statusType={areProjectsLoading ? 'loading' : 'finished'}
+            />
+          </FormField>
+        </SpaceBetween>
+      </ExpandableSection>
+    );
+  };
+
+  const getFileUploadSection = (): JSX.Element => {
+    return (
+      <ExpandableSection
+        defaultExpanded
+        variant="container"
+        header={
+          <Header
+            variant="h2"
+            description="Upload a file or folder associated with the Dataset being created or imported.  You can also add files from the Dashboard."
+          >
+            Upload files/folder - <i>optional</i>
+          </Header>
+        }
+      >
+        <SpaceBetween direction="vertical" size="l">
+          <FormField label="Upload files/folder" errorText={formErrors?.fileError}>
+            <FileUpload
+              data-testid="datasetFile"
+              value={formData?.file}
+              constraintText="All file type accepted. 5GB max file size."
+              accept=""
+              onChange={({ detail: { value } }) => {
+                setFormData({ ...formData, file: value });
+              }}
+            />
+          </FormField>
+        </SpaceBetween>
+      </ExpandableSection>
+    );
+  };
+
+  const getSharedProjectsSection = (): JSX.Element => {
+    return (
+      <ExpandableSection
+        defaultExpanded
+        variant="container"
+        header={
+          <Header
+            variant="h2"
+            description="Selected projects are visible to all users that are a part of the project."
+          >
+            Set permission to share project
+          </Header>
+        }
+      >
+        <TextContent>TODO: shared projects component</TextContent>
```

Comment:
Is this still needed?

Reply:
Yes, that was changed quite recently after implementation started. We have another story to handle this component